### DISCUSSION TITLE
conf.php not loading

### DISCUSSION THREAD

**Discussion by ghost** [PrivateBin/PrivateBin#1098] [Q&A] [answered]
Hello. I apologize for the second posting and issue. I am clearly doing something wrong here.

When I am using the non config run command below, everything works great using Cloudflare Tunnels behind https and under the appropriate subdomain. No errors.

docker run -d --restart="always" --read-only -p 8080:8080 -v $PWD/privatebin-data:/srv/data privatebin/nginx-fpm-alpine

When I am using the config run command below, I receive a Error 500.

docker run -d --restart="always" --read-only -p 8080:8080 -v $PWD/conf.php:/srv/cfg/conf.php:ro -v $PWD/privatebin-data:/srv/data privatebin/nginx-fpm-alpine

I did review the open / closed issues / FAQs / reddit / google etc. There seems to be a common thread about a PATH needing to be used. Below is one of the threads I found.

https://github.com/PrivateBin/PrivateBin/issues/746

I am new to docker etc. I am able to see the conf.php via SSH and in portainer. I just want to make adjustments to the retention rate of the pastes. Is someone able to point me in the correct direction please? I am guessing it has to be something with the PATH (for PHP/config?) I have seen in my searching and reading of the info I found here on the issues tab.

Thank you in advance. Have a good day.

**Comment by elrido** [comment]
Without having any insight into the log, we can only guess. It is likely that you simply have not given the UID/GID used in the container permission to read the conf.php file that you attach to the container. Check "docker logs <your container ID>" to learn what is actually going on that causes the error 500 - which can occur for all kinds of reasons.

**Reply by ghost** [reply]
Thank you.

Here is a copy of the logs I pulled from portainer.

[12-Jun-2023 14:25:23] NOTICE: fpm is running, pid 9
[12-Jun-2023 14:25:23] NOTICE: ready to handle connections
2023/06/12 14:28:42 [error] 18#18: *1 FastCGI sent in stderr: "PHP message: PHP Warning:  parse_ini_file(/srv/cfg/conf.php): Failed to open stream: No error information in /srv/lib/Configuration.php on line 119; PHP message: PHP Fatal error:  Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, bool given in /srv/lib/Configuration.php:121

Stack trace:
#0 /srv/lib/Configuration.php(121): array_key_exists()
#1 /srv/lib/Controller.php(164): PrivateBin\Configuration->__construct()
#2 /srv/lib/Controller.php(121): PrivateBin\Controller->_init()
#3 /var/www/index.php(18): PrivateBin\Controller->__construct()

}
  thrown in /srv/lib/Configuration.php on line 121" while reading response header from upstream, client: IP, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "subdomain.tld.org"
IP - - [12/Jun/2023:14:28:42 +0000] "GET / HTTP/1.1" 500 5 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36" "260:46:57:421:e194:883:aa6:ac1"

**Reply by ghost** [reply]
this is now resolve based on another discussion that was opened.

https://github.com/orgs/PrivateBin/discussions/1100#discussioncomment-6155599

**Comment by ghost** [comment, marked as answer]
https://github.com/orgs/PrivateBin/discussions/1100#discussioncomment-6155599
